server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -358,8 +359,18 @@ private void removeUnallocatedIndices(final ClusterChangedEvent event) {
                 assert indexMetaData != null || event.isNewCluster() :
                     "index " + index + " does not exist in the cluster state, it should either " +
                         "have been deleted or the cluster must be new";
-                final AllocatedIndices.IndexRemovalReason reason =
-                    indexMetaData != null && indexMetaData.getState() == IndexMetaData.State.CLOSE ? CLOSED : NO_LONGER_ASSIGNED;
+                AllocatedIndices.IndexRemovalReason reason = NO_LONGER_ASSIGNED;
+                if (indexMetaData != null) {
```

Comment:
I'm not sure if it's worth the complexity of this code here just to provide a better message as to why an index service got removed. If you think it's useful, maybe factor the logic of determining  the `AllocatedIndices.IndexRemovalReason` based on currentState and newState into a helper method so it can be reused by removeIndices